packages/react/src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -164,11 +164,31 @@ const Panel: React.FC<SelectPanelProps> = ({
   /* Dialog */
   const dialogRef = React.useRef<HTMLDialogElement>(null)
 
+  /* Anchored position */
+  const {position, updatePosition} = useAnchoredPosition(
+    {
+      anchorElementRef: anchorRef,
+      floatingElementRef: dialogRef,
+      side: 'outside-bottom',
+      align: 'start',
+    },
+    [
+      /* 
+        Because we call dialog.showModal() to open the dialog, 
+        we need to wait until the browser opens the dialog (giving it dimensions)
+        before updating position. (it is 0px by 0px until opened)
+        This is why we explicitly call updatePosition instead of using dependencies.
+      */
+    ],
+  )
+
   // sync dialog open state (imperative) with internal component state
   React.useEffect(() => {
-    if (internalOpen) dialogRef.current?.showModal()
-    else if (dialogRef.current?.open) dialogRef.current.close()
-  }, [internalOpen])
+    if (internalOpen) {
+      dialogRef.current?.showModal()
+      updatePosition() // update the position once the dialog is open
+    } else if (dialogRef.current?.open) dialogRef.current.close()
```

Comment:
This could skip the check

```suggestion
    } else dialogRef.current?.close()
```